tsd/tsd.go

```diff
@@ -76,6 +76,14 @@ type System struct {
 	userMetricsRegistry usermetric.Registry
 }
 
+// NewSystemWithEventBus constructs a new otherwise-empty system with a
+// freshly-constructed event bus populated.
+func NewSystemWithEventBus() *System {
+	sys := new(System)
+	sys.Set(eventbus.New())
+	return sys
+}
+
```

Comment:
This feels out of place. Why a constructor for just this one field? There are a dozen components that don't get this treatment.

I'd just remove this and let the caller do it.

Reply:
See the rest of the PR? We used `new(System)` all over the place with the assumption that it's a usable value, and we're explicitly changing that so that we don't have to defensively `if != nil` all over the place for something we want to make universal. I'd rather not force every future test that wants a tsd.System to segfault and discover that there's mandatory busywork it has to perform, no?

Reply:
If it's required, then do my second comment?

It's still weird to make a constructor that only sets 1 required thing when there are a half dozen required things.

Reply:
Or just make it `NewSystem` and not `NewSystemWithEventBus` at least.

Reply:
This is indeed a special case, and is meant as a temporary shim.

I don't think we need to leave it here permanently, but this value is meant to become "required", and I don't think we want to scatter the construction across multiple components.  Morally, we want to have one of these constructed in one place, ideally here. Having a special-purpose constructor is a little weird, but gives us an easy thing to grep for and replace later.

Reply:
> Or just make it `NewSystem` and not `NewSystemWithEventBus` at least.

Yeah, I considered that, but I didn't want to burn the "good" name on something that I strongly suspect will be temporary.

Reply:
Given that this is on a branch for now, I am inclined to leave it as-written, so that we can unblock some of the other changes further downstream. We can revisit this if we think it's important before we merge back into main after the release freeze.

Reply:
As an additional note for posterity, we also discussed the option of making a nil bus usable with no-op/discard semantics, but decided to defer that because it would make it hard to find places where we actually wanted a real value but can't tell statically that we missed it. We know we are going to need some more plumbing for tests too, but would rather discover that.